### PR TITLE
nucliadb_texts integration tests

### PR DESCRIPTION
**Pull request by jotare**
### Description
- Move integration tests from `reader.rs` and `writer.rs` to proper integration tests folder.
- Split tests for better readability 

### How was this PR tested?
- This tests

### CODE REVIEW

File: nucliadb_texts/src/writer.rs [hunk 1]
```diff
@@ -187,11 +169,22 @@ impl TextWriterService {
...
+            }),
+            ..Default::default()
+        };
+        let field_schema = TextSchema::new();
+        let mut index_builder = Index::builder().schema(field_schema.schema.clone());
+        index_builder = index_builder.settings(settings);
+        let index = index_builder
+            .create_in_dir(&config.path)
+            .expect("Index directory should exist");
         let writer = index.writer_with_num_threads(1, 6_000_000).unwrap();
```
Comment (tarekziade) [inline nucliadb_texts/src/writer.rs:188]: fly-by: `6_000_000` should be a const

File: nucliadb_texts/src/writer.rs [hunk 2]
```diff
@@ -201,6 +194,24 @@ impl TextWriterService {
...
     }
 
+    fn try_create_index_dir(path: &Path) -> NodeResult<()> {
+        let result = fs::create_dir(path);
+        if let Err(error) = result {
+            if path.exists() {
+                // operation failed but directory exists, we must delete it
+                if let Err(remove_error) = std::fs::remove_dir(path) {
+                    return Err(node_error!(
+                        "Double error creating and removing texts directory: \nFirst: {error} \
```
Comment (tarekziade) [inline nucliadb_texts/src/writer.rs:204]: is the extra `\` intended?

**Review by tarekziade — APPROVED** [review]
just nits

File: nucliadb_texts/src/writer.rs (hunk 2 above)
Comment (jotare) [reply nucliadb_texts/src/writer.rs:204]: Yes, it comes from `rustfmt` splitting the string

File: nucliadb_texts/src/writer.rs (hunk 1 above)
Comment (jotare) [reply nucliadb_texts/src/writer.rs:188]: Agree, I'll fix it in the next one

**Pull request merged**